packager/live_packager.cc

```diff
@@ -0,0 +1,163 @@
+// Copyright 2020 Google LLC. All rights reserved.
+//
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file or at
+// https://developers.google.com/open-source/licenses/bsd
+
+#include <fstream>
+#include <sstream>
+
+#include <packager/packager.h>
+#include <packager/file/file.h>
+#include <packager/file/file_closer.h>
+#include <packager/live_packager.h>
+#include "live_packager.h"
+
+namespace shaka {
+
+#define TAG(a, b, c, d)                                               \
+  ((static_cast<uint32_t>(static_cast<uint8_t>(a)) << 24) |           \
+   (static_cast<uint8_t>(b) << 16) | (static_cast<uint8_t>(c) << 8) | \
+   (static_cast<uint8_t>(d)))
+
+Segment::Segment(const uint8_t *data, size_t size) 
+    : data_(data, data + size) {}
```

Comment:
Do we really want to copy data here?  seems like we can use std::span(since C++20) to just hold a pointer and a size.